src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,144 @@
+                                Func<ChannelOptions, Channel<int>> bounded = o => Channel.CreateBounded<int>((BoundedChannelOptions)o);
+                                yield return new object[] { bounded, new BoundedChannelOptions(capacity)
+                                                                        {
+                                                                            SingleReader = singleReader,
+                                                                            SingleWriter = singleWriter,
+                                                                            AllowSynchronousContinuations = allowSynchronousContinuations,
+                                                                            FullMode = bco
+                                                                        }};
+                            }
+
+        }
+
+        const int MaxNumberToWriteToChannel = 1_000_000;
+        private static readonly int MaxTaskCounts = Math.Max(2, Environment.ProcessorCount);
+
+        [OuterLoop]
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public void RunInStressMode(Func<ChannelOptions, Channel<int>> channelCreator, ChannelOptions options)
+        {
+            Channel<int> channel = channelCreator(options);
+            ChannelReader<int> reader = channel.Reader;
+            ChannelWriter<int> writer = channel.Writer;
+            BoundedChannelOptions boundedOptions = options as BoundedChannelOptions;
+            bool shouldReadAllWrittenValues = boundedOptions == null || boundedOptions.FullMode == BoundedChannelFullMode.Wait;
+
+            List<Task> taskList = new List<Task>();
+
+            int readerTasksCount;
+            int writerTasksCount;
+
+            if (options.SingleReader)
+            {
+                readerTasksCount = 1;
+                writerTasksCount = options.SingleWriter ? 1 : MaxTaskCounts - 1;
+            }
+            else if (options.SingleWriter)
+            {
+                writerTasksCount = 1;
+                readerTasksCount = MaxTaskCounts - 1;
+            }
+            else
+            {
+                readerTasksCount = MaxTaskCounts / 2;
+                writerTasksCount = MaxTaskCounts - readerTasksCount;
+            }
+
+            int readCount = 0;
+
+            for (int i=0; i < readerTasksCount; i++)
+            {
+                taskList.Add(Task.Run(async delegate
+                {
+                    while (true)
+                    {
+                        try
+                        {
+                            if (!reader.TryRead(out int value))
+                            {
+                                value = await reader.ReadAsync();
+                            }
+                            Interlocked.Increment(ref readCount);
+                        }
+                        catch (ChannelClosedException)
+                        {
+                            break;
```

Comment:
You can move the try/catch to be outside of the while loop.  Then this break isn't necessary, and we're also not entering/leaving the try on each iteration.

Reply:
I'll do that although the intention of this test is to optimize the execution. it is just a stress test which should be doing some not desired patterns too.